bin/ebuild.sh

```diff
@@ -242,34 +242,32 @@ inherit() {
 		debug-print "*** Multiple Inheritence (Level: ${ECLASS_DEPTH})"
 	fi
 
-	if [[ -n $ECLASS && -n ${!__export_funcs_var} ]] ; then
-		echo "QA Notice: EXPORT_FUNCTIONS is called before inherit in" \
-			"$ECLASS.eclass. For compatibility with <=portage-2.1.6.7," \
-			"only call EXPORT_FUNCTIONS after inherit(s)." \
-			| fmt -w 75 | while read -r ; do eqawarn "$REPLY" ; done
-	fi
-
+	local -x ECLASS
+	local __export_funcs_var
 	local repo_location
 	local location
 	local potential_location
 	local x
-
-	# These variables must be restored before returning.
-	local PECLASS=$ECLASS
-	local prev_export_funcs_var=$__export_funcs_var
-
 	local B_IUSE
 	local B_REQUIRED_USE
 	local B_DEPEND
 	local B_RDEPEND
 	local B_PDEPEND
 	local B_HDEPEND
 	local B_BDEPEND
+
+	if [[ -n ${ECLASS} && -n ${!__export_funcs_var} ]] ; then
+		echo "QA Notice: EXPORT_FUNCTIONS is called before inherit in" \
+			"${ECLASS}.eclass. For compatibility with <=portage-2.1.6.7," \
+			"only call EXPORT_FUNCTIONS after inherit(s)." \
+			| fmt -w 75 | while read -r ; do eqawarn "${REPLY}" ; done
```

Comment:
This code was just moved in this commit, but I will note that it does not work since commit fb2459330cf226ee34d3875a1143531bd109aaf2 :) .
`fmt` is `/usr/bin/fmt`...

Reply:
Thanks, fixed in 2d37467170737bf055a058322c543f313737b983.

Reply:
Yes it would be nice to drop the warning, but the current  definition of EXPORT_FUNCTIONS in PMS is not really specific about this case.